File: ChatGPT_BP_PY/CWE-416_BP/416_1b_BP.py
import sys

def main():
    MAX_BUFFER_SIZE = 1024  # Define a reasonable maximum buffer size

    # Check if exactly one command-line argument is provided
    if len(sys.argv) != 2:
        print("Usage: python3 buffer_allocator.py <buffer_size>")
        sys.exit(1)

    try:
        # Convert the input to an integer
        buffer_size = int(sys.argv[1])

        # Validate the buffer size
        if buffer_size <= 0:
            raise ValueError("Buffer size must be a positive integer.")
        if buffer_size > MAX_BUFFER_SIZE:
            raise ValueError(f"Buffer size must not exceed {MAX_BUFFER_SIZE}.")

        # Allocate buffer
        buffer = [0] * buffer_size

        # Print buffer contents
        print(f"Buffer of size {buffer_size} allocated.")
        print(buffer)

    except ValueError as ve:
        # Handle invalid integer and range errors
        print(f"Error: {ve}")
        sys.exit(1)
    except Exception as e:
        # Handle other unexpected errors
        print(f"Unexpected error: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main()

File: ChatGPT_BP_Java/CWE-416/416_1b_BP.java
import java.util.InputMismatchException;

public class SecureBuffer {
    
    private static final int MAX_BUFFER_SIZE = 1024; // Define a reasonable maximum buffer size

    public static void main(String[] args) {
        // Validate input
        if (args.length != 1) {
            System.err.println("Usage: java SecureBuffer <buffer_size>");
            System.exit(1);
        }

        int bufferSize = 0;

        try {
            // Parse and validate buffer size
            bufferSize = Integer.parseInt(args[0]);

            // Check buffer size range
            if (bufferSize <= 0 || bufferSize > MAX_BUFFER_SIZE) {
                throw new IllegalArgumentException("Buffer size must be a positive integer and less than or equal to " + MAX_BUFFER_SIZE);
            }
        } catch (NumberFormatException | IllegalArgumentException e) {
            System.err.println("Invalid buffer size: " + e.getMessage());
            System.exit(1);
        }

        // Securely allocate the buffer
        char[] buffer = new char[bufferSize];
        
        // Securely initialize the buffer (fill with a safe value)
        for (int i = 0; i < buffer.length; i++) {
            buffer[i] = '\u0000'; // Initialize to null character
        }

        // Print the contents of the buffer safely
        System.out.println("Buffer contents: " + new String(buffer));

        // Securely clear the buffer before deallocation
        clearBuffer(buffer);

        // Optionally, set to null for garbage collection (not strictly necessary)
        buffer = null;
    }

    // Securely clear the buffer to prevent sensitive data leakage
    private static void clearBuffer(char[] buffer) {
        if (buffer != null) {
            for (int i = 0; i < buffer.length; i++) {
                buffer[i] = '\u0000'; // Overwrite with null character
            }
        }
    }
}
